src/main/java/umk/mat/pajda/ProjektZespolowy/services/OpinionService.java

```diff
@@ -0,0 +1,33 @@
+package umk.mat.pajda.ProjektZespolowy.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import umk.mat.pajda.ProjektZespolowy.DTO.OpinionPostDTO;
+
+@Service
+public class OpinionService {
+
+  private final ReviewService reviewService;
+
+  @Autowired(required = false)
+  private TipService tipService;
+
+  private final Logger logger = LoggerFactory.getLogger(OpinionService.class);
+
+  public OpinionService(ReviewService reviewService) {
+    this.reviewService = reviewService;
+  }
+
+  public ResponseEntity<String> addOpinion(OpinionPostDTO opinionPostDTO, String ip) {
```

Comment:
Shouldn't it return bool , because in other services methods we usually return bool when we don't intend to return a data to user

Reply:
We return response from PayU as you can see in TipService. So setting boolean will lose data

Reply:
ok